src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/LocalParamServer.java

```diff
@@ -24,36 +24,37 @@
 import org.apache.sysml.runtime.controlprogram.context.ExecutionContext;
 import org.apache.sysml.runtime.instructions.cp.Data;
 import org.apache.sysml.runtime.instructions.cp.ListObject;
+import org.apache.sysml.runtime.instructions.cp.ParamservBuiltinCPInstruction;
 
 public class LocalParamServer extends ParamServer {
 
 	public LocalParamServer(ListObject model, String aggFunc, Statement.PSFrequency freq,
 			Statement.PSUpdateType updateType, ExecutionContext ec, int workerNum,
-			ListObject hyperParams) {
-		super(model, aggFunc, freq, updateType, ec, workerNum, hyperParams);
+			ParamservBuiltinCPInstruction.PSErrorHandler handler) {
+		super(model, aggFunc, freq, updateType, ec, workerNum, handler);
 	}
 
 	@Override
 	public void push(long workerID, ListObject gradients) {
-		synchronized (_lock) {
-			_queue.add(new Gradient(workerID, gradients));
-			_lock.notifyAll();
+		try {
+			_gradientsQueue.put(new Gradient(workerID, gradients));
+			launchService();
+		} catch (InterruptedException e) {
+			throw new DMLRuntimeException(
+					String.format("Local param server: failed to push the gradients for worker_%d.", workerID), e);
 		}
```

Comment:
I leave it up to you but there is no need for explicit exception handling for each queue interaction. You have a surrounding exception handling and the issue would simply show up in the stack trace.